pkg/integrations/v2/app_agent_receiver/app_agent_receiver.go

```diff
@@ -0,0 +1,193 @@
+package app_agent_receiver //nolint:golint
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/gorilla/mux"
+	"github.com/grafana/agent/pkg/integrations/v2"
+	"github.com/grafana/agent/pkg/integrations/v2/metricsutils"
+	"github.com/grafana/agent/pkg/traces/pushreceiver"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/weaveworks/common/instrument"
+	"github.com/weaveworks/common/middleware"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+)
+
+type appAgentReceiverIntegration struct {
+	integrations.MetricsIntegration
+	appAgentReceiverHandler AppAgentReceiverHandler
+	logger                  log.Logger
+	conf                    *Config
+	reg                     *prometheus.Registry
+
+	requestDurationCollector     *prometheus.HistogramVec
+	receivedMessageSizeCollector *prometheus.HistogramVec
+	sentMessageSizeCollector     *prometheus.HistogramVec
+	inflightRequestsCollector    *prometheus.GaugeVec
+}
+
+// Static typecheck tests
+var (
+	_ integrations.Integration        = (*appAgentReceiverIntegration)(nil)
+	_ integrations.HTTPIntegration    = (*appAgentReceiverIntegration)(nil)
+	_ integrations.MetricsIntegration = (*appAgentReceiverIntegration)(nil)
+)
+
+// NewIntegration converts this config into an instance of an integration
+func (c *Config) NewIntegration(l log.Logger, globals integrations.Globals) (integrations.Integration, error) {
+	reg := prometheus.NewRegistry()
+	sourcemapLogger := log.With(l, "subcomponent", "sourcemaps")
+	sourcemapStore := NewSourceMapStore(sourcemapLogger, c.SourceMaps, reg, nil, nil)
+
+	receiverMetricsExporter := NewReceiverMetricsExporter(ReceiverMetricsExporterConfig{
```

Comment:
nit: IMO, I don't think anything in this package aside from the config structs need to be exported 🤷

Reply:
I think there could be cases where we want to import these in another project. For example implementing a reicever endpoint in Grafana's backend, so that ootb it could receive and log frontend events w/o running an agent additionally.